src/ui/server/controllers/insurance/your-buyer/currency-of-late-payments/save-and-back/index.ts

```diff
@@ -0,0 +1,53 @@
+import { INSURANCE_ROUTES } from '../../../../../constants/routes/insurance';
+import hasFormData from '../../../../../helpers/has-form-data';
+import { FIELD_IDS } from '..';
+import constructPayload from '../../../../../helpers/construct-payload';
+import generateValidationErrors from '../validation';
+import mapAndSave from '../../map-and-save/buyer-trading-history';
+import { Request, Response } from '../../../../../../types';
+
+const { INSURANCE_ROOT, ALL_SECTIONS, PROBLEM_WITH_SERVICE } = INSURANCE_ROUTES;
+
+/**
+ * post
+ * Save any valid currency of late payments form fields and if successful, redirect to the all sections page
+ * @param {Express.Request} Express request
+ * @param {Express.Response} Express response
+ * @returns {Express.Response.redirect} All sections page or error page
+ */
+export const post = async (req: Request, res: Response) => {
+  try {
+    const { application } = res.locals;
+
+    if (!application) {
+      return res.redirect(PROBLEM_WITH_SERVICE);
+    }
+
+    const { referenceNumber } = req.params;
+
+    /**
+     * If form data is populated:
+     * 1) generate a payload.
+     * 2) generate validation errors.
+     * 3) call mapAndSave
+     * 4) redirect
+     */
+    if (hasFormData(req.body)) {
+      const payload = constructPayload(req.body, FIELD_IDS);
+
+      const validationErrors = generateValidationErrors(payload);
+
+      const saveResponse = await mapAndSave.buyerTradingHistory(payload, application, validationErrors);
+
+      if (!saveResponse) {
+        return res.redirect(PROBLEM_WITH_SERVICE);
+      }
+    }
+
+    return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${ALL_SECTIONS}`);
+  } catch (error) {
+    console.error('Error updating application - your buyer - currency of late payments (save and back) %O', error);
```

Comment:
I know there is a PR for changing these all to %o but could the new pages not have %o so then we change it in less pages later?

Reply:
Agree - this PR was created after we discovered this needs to change. WIll update.